Source/multi.cpp

```diff
@@ -598,7 +603,7 @@ void multi_process_network_packets()
 		ClearPlayerLeftState();
 		if (dwMsgSize < sizeof(TPktHdr))
 			continue;
-		if (dwID < 0 || dwID >= MAX_PLRS)
+		if (static_cast<size_t>(dwID) >= Players.size())

```

Comment:
The underlying type being used in SNetReceiveMessage is uint8_t, so lets change dwID to that as well and set the default to max.